app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/SenderHandShake.kt

```diff
@@ -0,0 +1,148 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.localFileTransfer
+
+import android.util.Log
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.withContext
+import org.kiwix.kiwixmobile.core.BuildConfig
+import java.io.InputStream
+import java.io.ObjectInputStream
+import java.io.ObjectOutputStream
+import java.io.OutputStream
+import java.net.InetAddress
+import java.net.InetSocketAddress
+import java.net.ServerSocket
+import java.net.Socket
+
+class SenderHandShake :
```

Comment:
this is just the original handshake task, there has been no separation of concerns.

Right now we have

```
if(groupOwner){
  readHandshake()
  if(isSender) sendMetadata()
  else readMetadata()
} else {
  sendHandshake()
  if(isSender) sendMetadata()
  else readMetadata()
}
```

Have we confirmed the relationship between groupowner and sender/reciever? Is the owner always the sender? Is the owner always the receiver? Does it depend on some external factor?

if it was always the sender then we would have a class SenderHandshake like

```
readHandshake()
sendMetadata()
```
with a ReceiverHandshake like
```
writeHandshake()
receiveMetadata()
```


if it was always the receiver then we would have a class ReceiverHandshake like
```
readHandshake()
receiveMetadata()
```

with a SenderHandshake like
```
writeHandshake()
sendMetadata()
```